### PR TITLE
Prohibit clients from declaring placeholders

### LINKED ISSUE #2010: Is SETTINGS_NUM_PLACEHOLDERS sent by client an error?
Should receipt of SETTINGS_NUM_PLACEHOLDERS by the server trigger an error?

### PR DESCRIPTION
**Pull request by MikeBishop**
Fixes #2010.

### CODE REVIEW

**Comment by martinthomson** [comment]
OMFG, moar erroar cohdz.

File: draft-ietf-quic-http.md [hunk 1]
```diff
@@ -1028,7 +1028,9 @@ the prioritization the client had attempted to express.
 In HTTP/3, a number of placeholders are explicitly permitted by the server using
 the `SETTINGS_NUM_PLACEHOLDERS` setting. Because the server commits to maintain
```
Comment (janaiyengar) [inline draft-ietf-quic-http.md]: s/maintain/maintaining

**Comment by janaiyengar** [comment]
Heh. I like that 0x01 is getting used now, but can't you use UNEXPECTED_FRAME for this?

**Comment by MikeBishop** [comment]
It's a setting, not a frame.  We could expand the scope of that error code, if folks think that's preferable.

**Pull request merged**